pouch/convertor/cri_test.go

```diff
@@ -0,0 +1,309 @@
+				"io.kubernetes.pouch.type":                            "sandbox",
+			},
+		},
+		HostConfig: &pouchtypes.HostConfig{
+			Runtime:     "runc",
+			EnableLxcfs: false,
+			Resources: pouchtypes.Resources{
+				CgroupParent: "/kubepods/besteffort/pod4802cf32-f21e-11e8-b9c9-6c92bf2cf061",
+			},
+		},
+		State: &pouchtypes.ContainerState{
+			Pid: int64(159406),
+		},
+		NetworkSettings: &pouchtypes.NetworkSettings{
+			Networks: map[string]*pouchtypes.EndpointSettings{
+				"none": {
+					EndpointID: "22e5df9fc43a97cc3f65fe89d18ebcc5cec2b1c6e33c170dda69922a705aad26",
+					NetworkID:  "5f834b062553a2d9b8a91406fc2e75ea571f74a4b0c9bbe3666722a9e18c2f89",
+				},
+			},
+		},
+	}
+
+	wantCRIMeta := localtypes.SandboxMeta{
+		ID: "ce8999940e12e653d4308cc89a18e25fcf702dea250fc2428299bed5dfb03e92",
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      "nginx",
+				Uid:       "4802cf32-f21e-11e8-b9c9-6c92bf2cf061",
+				Namespace: "default",
+			},
+			Hostname:     "nginx",
+			LogDirectory: "/var/log/pods/4802cf32-f21e-11e8-b9c9-6c92bf2cf061",
+			Labels: map[string]string{
+				"ali/test":                    "123456",
+				"io.kubernetes.pod.name":      "nginx",
+				"io.kubernetes.pod.namespace": "default",
+				"io.kubernetes.pod.uid":       "4802cf32-f21e-11e8-b9c9-6c92bf2cf061",
+			},
+			Annotations: map[string]string{
+				"kubernetes.io/config.seen":                "2018-11-27T16:27:30.623885666+08:00",
+				"kubernetes.io/config.source":              "api",
+				"pod.beta1.sigma.ali/container-state-spec": "{\"states\":{\"nginx\":\"running\"}}",
+				"pod.beta1.sigma.ali/pod-spec-hash":        "1122abcd",
+			},
+			Linux: &runtime.LinuxPodSandboxConfig{
+				CgroupParent: "/kubepods/besteffort/pod4802cf32-f21e-11e8-b9c9-6c92bf2cf061",
+				SecurityContext: &runtime.LinuxSandboxSecurityContext{
+					NamespaceOptions: &runtime.NamespaceOption{
+						Pid: runtime.NamespaceMode_CONTAINER,
+					},
+				},
+			},
+		},
+		Runtime:      "runc",
+		LxcfsEnabled: false,
+		NetNS:        "/proc/159406/ns/net",
+	}
+
+	gotCRIMeta, err := ToCRIMetaJSON(testCon)
+	if err != nil {
+		t.Fatalf("failed to generate CRI meta: %v", err)
+	}
+
+	// check general parameters
+	if gotCRIMeta.ID != wantCRIMeta.ID || gotCRIMeta.Runtime != wantCRIMeta.Runtime || gotCRIMeta.LxcfsEnabled != wantCRIMeta.LxcfsEnabled {
+		t.Errorf("wanted to (ID: %v, Runtime: %v, LxcfsEnabled: %v), got (%v, %v, %v)", wantCRIMeta.ID, wantCRIMeta.Runtime, wantCRIMeta.LxcfsEnabled, gotCRIMeta.ID, gotCRIMeta.Runtime, gotCRIMeta.LxcfsEnabled)
+	}
+
+	// check PodSandboxConfig
+	if !reflect.DeepEqual(wantCRIMeta.Config, gotCRIMeta.Config) {
+		t.Errorf("got wrong PodSandboxConfig, want %v, got %v", wantCRIMeta.Config, gotCRIMeta.Config)
+	}
+}
+
+func Test_toDNSConfig(t *testing.T) {
+	var (
+		resolvContent = `search hz.ali.com
+nameserver 10.65.1.3
+nameserver  30.14.130.2
+nameserver 10.65.1.1
```

Comment:
I am afraid that there would be some security issue. just using fake ip in test.